### ISSUE TITLE
Macro-provided `body` conflicts with `body` in extensions

### ISSUE THREAD

**Issue by andtie** [pointfreeco/swift-composable-architecture#2817]
### Description

In our complex app, we have some pretty big features. Therefore, we have sometimes moved the reducer-body into a different file as an extension of the feature. This triggers the following bug:

If the `body ` of a `Reducer` is provided in an extension, the `@Reducer`macro does not see it and creates a conflicting empty `body`.

Example:
```swift
@Reducer 
struct TestFeature {
    struct State: Equatable {
        var count: Int
    }

    enum Action {
        case increment, decrement
    }

// Generated by the macro:
//let body = ComposableArchitecture.EmptyReducer<State, Action>()`
}

// In a different file TestFeature+Reducer:
extension TestFeature {
    @ReducerBuilder<State, Action> // we have to explicitly add the ReducerBuilder, because the default-value from the protocol does not work with extensions. See also https://github.com/apple/swift/issues/65252
    var body: some Reducer<State, Action> {
        Reduce { state, action in
            switch action {
            case .increment:
                state.count += 1
            case .decrement:
                state.count -= 1
            }
            return .none
        }
    }
}
```

I am aware that a macro cannot see extensions and that there are other workarounds for splitting a large feature. Nevertheless, I think this is a source-breaking change that you should be aware of and that maybe can be mitigated.

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

The code should compile. 

### Actual behavior

The following is generated by the macro:
`let body = ComposableArchitecture.EmptyReducer<State, Action>()`
This leads to the following error:
> ❌ Invalid redeclaration of 'body'

### The Composable Architecture version information

1.8.0 / main

### Destination operating system

iOS 17.3.1

### Xcode version information

Version 15.2 (15C500b)

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @andtie, thanks for bringing this up. I think the only way to support this is to provide a new argument to `@Reducer` that suppresses the synthesized requirements:

```swift
@Reduer(synthesizeRequirements: false)
struct Feature {
  …
}
```

We'll think about it a bit more.

**Comment by mbrandonw** [comment]
Hi @andtie, we discussed this today and we have decided to leave it as-is. If you want to do something specialized like move `State`, `Action` or `body` to a different file, then you will just need to forgo using `@Reducer` and implement the details manually. That's just the reality of dealing with macros.

Sorry that this breaking change snuck into a minor release. It's very subtle and we didn't realize it at the time.

I am going to close this issue now, but feel free to open a discussion if you want to discuss further.
